src/coreComponents/constitutive/contact/CoulombContact.hpp

```diff
@@ -219,11 +232,78 @@ void CoulombContactUpdates::computeTraction( localIndex const k,
-    real64 const slipNorm = LvArray::tensorOps::l2Norm< 2 >( slip );
+
 
     real64 const tau[2] = { m_shearStiffness * ( slip[0] + m_elasticSlip[k][0] ),
                             m_shearStiffness * ( slip[1] + m_elasticSlip[k][1] ) };
 
+    switch( fractureState )
+    {
+      case extrinsicMeshData::contact::FractureState::Stick:
+      {
+        // Elastic slip case
+        // Tangential components of the traction are equal to tau
+        tractionVector[1] = tau[0];
+        tractionVector[2] = tau[1];
+
+        dTractionVector_dJump[1][1] = m_shearStiffness;
+        dTractionVector_dJump[2][2] = m_shearStiffness;
+
+        // The slip is only elastic: we add the full slip to the elastic one
+        LvArray::tensorOps::add< 2 >( m_elasticSlip[k], slip );
+
+        break;
+      }
+      case extrinsicMeshData::contact::FractureState::Slip:
+      {
+        // Plastic slip case
+        real64 dLimitTau_dNormalTraction;
+        real64 const limitTau = computeLimitTangentialTractionNorm( tractionVector[0],
+                                                                    dLimitTau_dNormalTraction );
+
+        real64 const slipNorm = LvArray::tensorOps::l2Norm< 2 >( slip );
+
+        // Tangential components of the traction computed based on the limitTau
+        tractionVector[1] = limitTau * slip[0] / slipNorm;
+        tractionVector[2] = limitTau * slip[1] / slipNorm;
+
+        dTractionVector_dJump[1][0] = m_penaltyStiffness * dLimitTau_dNormalTraction * slip[0] / slipNorm;
+        dTractionVector_dJump[1][1] = limitTau * pow( slip[1], 2 )  / pow( LvArray::tensorOps::l2NormSquared< 2 >( slip ), 1.5 );
+        dTractionVector_dJump[1][2] = limitTau * slip[0] * slip[1] / pow( LvArray::tensorOps::l2NormSquared< 2 >( slip ), 1.5 );
+
+        dTractionVector_dJump[2][0] = m_penaltyStiffness * dLimitTau_dNormalTraction * slip[1] / slipNorm;
+        dTractionVector_dJump[2][1] = limitTau * slip[0] * slip[1] / pow( LvArray::tensorOps::l2NormSquared< 2 >( slip ), 1.5 );
+        dTractionVector_dJump[2][2] = limitTau * pow( slip[0], 2 )  / pow( LvArray::tensorOps::l2NormSquared< 2 >( slip ), 1.5 );
+
+        // Compute elastic component of the slip for this case
+        real64 const plasticSlip[2] = { tractionVector[1] / m_shearStiffness,
+                                        tractionVector[2] / m_shearStiffness };
+
+        LvArray::tensorOps::copy< 2 >( m_elasticSlip[k], slip );
+        LvArray::tensorOps::subtract< 2 >( m_elasticSlip[k], plasticSlip );
+
+        break;
+      }
+    }
+  }
+}
+
+GEOSX_HOST_DEVICE
+void CoulombContactUpdates::updateFractureState( localIndex const k,
+                                                 arraySlice1d< real64 const > const & dispJump,
+                                                 arraySlice1d< real64 const > const & tractionVector,
+                                                 integer & fractureState ) const
```

Comment:
@TotoGaz if done in scope this is fine even in an hpp file, right?

Reply:
I think it's OK if the `using namespace` dies when out of scope, yer.
I remember that @klevzoff mentioned something about it? Is there some specific care?

Reply:
I'm relatively certain you'll need `inline` on the function body (but not force inline), otherwise the linker will complain about ODR violations